src/transform/src/index.ts

```diff
@@ -3,8 +3,334 @@ import { createWriteStream } from "fs";
 import { HypermodeMetadata } from "./metadata.js";
 import { Extractor } from "./extractor.js";
 import binaryen from "assemblyscript/lib/binaryen.js";
+import {
+  ArrayLiteralExpression,
+  BlockStatement,
+  CommonFlags,
+  FunctionDeclaration,
+  IdentifierExpression,
+  IntegerLiteralExpression,
+  LiteralExpression,
+  LiteralKind,
+  NamedTypeNode,
+  NewExpression,
+  Node,
+  NodeKind,
+  ObjectLiteralExpression,
+  ParameterKind,
+  Parser,
+  Range,
+  Source,
+  SourceKind,
+  StringLiteralExpression,
+  Token,
+  Tokenizer,
+} from "assemblyscript/dist/assemblyscript.js";
+import { FunctionSignature, Parameter, TypeInfo } from "./types.js";
 
 export default class HypermodeTransform extends Transform {
+  public funcMetadata = new Map<string, FunctionSignature>();
```

Comment:
Can we move some of this new code to another file?  It's pretty big, and specific to this task.  Thanks.